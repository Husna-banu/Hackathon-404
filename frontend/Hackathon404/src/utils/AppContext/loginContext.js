import React from 'react'
const accessStateContext = React.createContext()
const accessDispatchContext = React.createContext()

function AccessReducer(state, action) {
  switch (action.type) {
    case 'login': {
      return { loginName: action.payload, login: true }
    }
    case 'logout': {
      return { loginName: action.payload, login: false }
    }
    default: {
      throw new Error(`Unhandled action type: ${action.type}`)
    }
  }
}

function AccessProvider({ children }) {
  const [state, dispatch] = React.useReducer(AccessReducer, { loginName: '', login: false })
  return (
    <accessStateContext.Provider value={state}>
      <accessDispatchContext.Provider value={dispatch}>
        {children}
      </accessDispatchContext.Provider>
    </accessStateContext.Provider>
  )
}

function useAccessState() {
  const context = React.useContext(accessStateContext)
  if (context === undefined) {
    throw new Error('useAccessState must be used within a AccessProvider')
  }
  return context
}

function useAccessDispatch() {
  const context = React.useContext(accessDispatchContext)
  if (context === undefined) {
    throw new Error('useAccessDispatch must be used within a AccessProvider')
  }
  return context
}

export { AccessProvider, useAccessState, useAccessDispatch }
