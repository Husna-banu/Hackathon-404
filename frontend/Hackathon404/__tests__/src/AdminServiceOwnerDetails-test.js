import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AdminServiceOwnerDetails from '../../src/Admin/ServicesOwnerDetails';
import {AccessProvider} from '../../src/utils/AppContext/loginContext';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
fetch = jest.fn(() => Promise.resolve());
const props = {
  navigate: jest.fn(),
  route: {
    params: {},
  },
};

it('renders correctly', async () => {
  const container = renderer.create(
    <AccessProvider>
      <AdminServiceOwnerDetails {...props} />
    </AccessProvider>,
  );
  expect(container).toMatchSnapshot();
});
