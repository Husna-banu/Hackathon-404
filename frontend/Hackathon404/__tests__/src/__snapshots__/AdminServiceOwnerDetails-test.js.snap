// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
      "top": 40,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "skyblue",
          "flexDirection": "row",
          "height": 50,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            Object {
              "fontSize": 20,
              "paddingLeft": 10,
              "paddingRight": 20,
            },
            Object {
              "fontFamily": "FontAwesome5Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "900",
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "alignItems": "center",
            "fontSize": 20,
            "fontWeight": "bold",
            "justifyContent": "center",
            "paddingLeft": 10,
            "textAlign": "center",
          }
        }
      >
         Services Owner Details
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 25,
            },
            Object {
              "position": "absolute",
              "right": 60,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <RCTScrollView
        data={
          Array [
            Object {
              "contactNo": 9173892005,
              "serviceName": "Restaurant",
              "serviceOwnerName": "Ram Kumar",
            },
            Object {
              "contactNo": 9173892004,
              "serviceName": "Spa",
              "serviceOwnerName": "Shyam Kumar",
            },
            Object {
              "contactNo": 9173892003,
              "serviceName": "Laundary",
              "serviceOwnerName": "Rahul Kumar",
            },
            Object {
              "contactNo": 9173892002,
              "serviceName": "Emergency Service",
              "serviceOwnerName": "Rohit Kumar",
            },
            Object {
              "contactNo": 9173892000,
              "serviceName": "Room Maintenance",
              "serviceOwnerName": "Raghu k",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#CDE6FA",
                  "borderColor": "gray",
                  "borderWidth": 1,
                  "justifyContent": "flex-start",
                  "marginBottom": 10,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "padding": 5,
                  }
                }
              >
                Restaurant
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                Ram Kumar
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                9173892005
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 25,
                    },
                    Object {
                      "position": "absolute",
                      "right": 40,
                      "top": 20,
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "position": "absolute",
                      "right": 10,
                      "top": 20,
                    },
                    Object {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#CDE6FA",
                  "borderColor": "gray",
                  "borderWidth": 1,
                  "justifyContent": "flex-start",
                  "marginBottom": 10,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "padding": 5,
                  }
                }
              >
                Spa
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                Shyam Kumar
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                9173892004
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 25,
                    },
                    Object {
                      "position": "absolute",
                      "right": 40,
                      "top": 20,
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "position": "absolute",
                      "right": 10,
                      "top": 20,
                    },
                    Object {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#CDE6FA",
                  "borderColor": "gray",
                  "borderWidth": 1,
                  "justifyContent": "flex-start",
                  "marginBottom": 10,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "padding": 5,
                  }
                }
              >
                Laundary
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                Rahul Kumar
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                9173892003
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 25,
                    },
                    Object {
                      "position": "absolute",
                      "right": 40,
                      "top": 20,
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "position": "absolute",
                      "right": 10,
                      "top": 20,
                    },
                    Object {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#CDE6FA",
                  "borderColor": "gray",
                  "borderWidth": 1,
                  "justifyContent": "flex-start",
                  "marginBottom": 10,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "padding": 5,
                  }
                }
              >
                Emergency Service
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                Rohit Kumar
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                9173892002
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 25,
                    },
                    Object {
                      "position": "absolute",
                      "right": 40,
                      "top": 20,
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "position": "absolute",
                      "right": 10,
                      "top": 20,
                    },
                    Object {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#CDE6FA",
                  "borderColor": "gray",
                  "borderWidth": 1,
                  "justifyContent": "flex-start",
                  "marginBottom": 10,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "padding": 5,
                  }
                }
              >
                Room Maintenance
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                Raghu k
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                9173892000
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 25,
                    },
                    Object {
                      "position": "absolute",
                      "right": 40,
                      "top": 20,
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "position": "absolute",
                      "right": 10,
                      "top": 20,
                    },
                    Object {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</View>
`;
